src/main/scala/com/evolution/scache/LoadingCache.scala

```diff
@@ -169,46 +236,72 @@ private[scache] object LoadingCache {
                                       entryRef
                                         .get
                                         .flatMap {
-                                          case Left(_)      =>
-                                            0.tailRecM { counter =>
+                                          // Our deferred is still there
+                                          case _: EntryState.Loading[F, V] =>
+                                            0.tailRecM { counter1 =>
                                               ref
                                                 .access
                                                 .flatMap { case (entryRefs, set) =>
                                                   entryRefs
                                                     .get(key)
                                                     .fold {
+                                                      // Key was removed while we were loading,
+                                                      // so we are just propagating the error
                                                       error.raiseError[F, Either[Int, V]]
                                                     } {
+                                                      // The entry we added to the map is still there and unmodified,
+                                                      // so we can safely remove it and propagate the error,
+                                                      // without needing to release anything.
                                                       case `entryRef` =>
                                                         set(entryRefs - key).flatMap {
                                                           case true  =>
                                                             error.raiseError[F, Either[Int, V]]
                                                           case false =>
-                                                            (counter + 1)
+                                                            (counter1 + 1)
                                                               .asLeft[V]
                                                               .pure[F]
                                                         }
-                                                      case entryRef   =>
+                                                      // Another fiber replaced the `ref` we added to the map,
+                                                      // which shouldn't be possible, but if it happens,
+                                                      // we return their value if it's already computed,
+                                                      // or propagate our error if it's still loading,
```

Comment:
we just return `other` value in both cases `Value` and `Loading` cases, to be consistent with happy path

Reply:
yep, done